File: Etap_3/src/main/java/com/mock_compare/mock_compare/transformer/CarTransformer.java
package com.mock_compare.mock_compare.transformer;

import com.mock_compare.mock_compare.dto.CarDto;
import com.mock_compare.mock_compare.models.Car;
import org.springframework.beans.BeanUtils;

public class CarTransformer {

    public static Car convertToEntity(CarDto dto) {
        Car car = new Car();
        BeanUtils.copyProperties(dto, car);
        car.setSalesman("None");
        return car;
    }

    public static CarDto convertToDto(Car car) {
        CarDto dto = new CarDto();
        BeanUtils.copyProperties(car, dto);
        return dto;
    }
}

File: Etap_6/mock_compare/transformer/CarTransformer.cs
using mock_compare.Dto;
using mock_compare.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace mock_compare.transformer
{
    public class CarTransformer
    {
        public static Car convertToEntity(CarDto dto)
        {
            Car car = new Car();
            car.setBrand(dto.Getbrand());
            car.setId(dto.Getid());
            car.setIsAvailable(dto.GetisAvailable());
            car.setModel(dto.Getmodel());
            car.setSalesman("None");
            return car;
        }

        public static CarDto convertToDto(Car car)
        {
            CarDto dto = new CarDto();
            dto.Setbrand(car.getBrand());
            dto.Setid(car.getId());
            dto.SetisAvailable(car.getIsAvailable());
            dto.Setmodel(car.getModel());
            return dto;
        }
    }
}
